File: backend/hiFes/src/main/java/hiFes/hiFes/repository/UserRepository.java
package hiFes.hiFes.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import hiFes.hiFes.entity.User;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByEmail(String email);  // 이미 email 을 통해 생성된 사용자인지 체크
}

File: frontend/android/app/src/main/java/com/ssafy/hifes/data/repository/user/UserRepository.kt
package com.ssafy.hifes.data.repository.user

import com.ssafy.hifes.data.model.ErrorResponse
import com.ssafy.hifes.data.model.FcmTokenDto
import com.ssafy.hifes.data.model.LoginResponse
import com.ssafy.hifes.util.network.NetworkResponse
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.http.Body
import retrofit2.http.Part
import retrofit2.http.Query

interface UserRepository {

    suspend fun login(@Query("accessToken") accessToken: String): NetworkResponse<LoginResponse, ErrorResponse>

    suspend fun signUp(
        @Part("normalUserSignUpDto") normalUserSignUpDto: RequestBody,
        @Part image: MultipartBody.Part
    ): NetworkResponse<LoginResponse, ErrorResponse>

    suspend fun saveFcmToken(@Body fcmToken: FcmTokenDto): NetworkResponse<Boolean, ErrorResponse>

}